src/Server.php

```diff
@@ -157,11 +157,16 @@ public function handleRequest(ConnectionInterface $conn, RequestInterface $reque
                 return $this->writeError($conn, 400);
             }
 
-            $stream = new LengthLimitedStream($conn, $contentLength);
+            $stream = new LengthLimitedStream($stream, $contentLength);
         }
 
         $request = new Request($request, $stream);
 
+        // Forward occuring errors on to request
+        $stream->on('error', function ($error) use ($request) {
+            $request->emit('error', array($error));
+        });
```

Comment:
Must be slipped through in a rebase :see_no_evil: . It was already moved to the Request object. Removed it.